### PR TITLE
feat(feedback/issues): add sidebar to issue details UF empty state

### PR DESCRIPTION
**Pull request by michellewzhang**
- add a sidebar popup when "set up now" is clicked
- add crash report modal onboarding config for javascript with widget callout

SCR-20240305-ocgr

### CODE REVIEW

File: static/app/components/feedback/feedbackOnboarding/sidebar.tsx [hunk 1]
```diff
@@ -147,6 +148,8 @@ function OnboardingContent({currentProject}: {currentProject: Project}) {
   }>(jsFrameworkSelectOptions[0]);
 
   const defaultTab = 'npm';
+  const {location} = useRouteContext();
+  const crashReportOnboarding = location.hash === '#crashreport-sidequest';
```
Comment (michellewzhang) [inline static/app/components/feedback/feedbackOnboarding/sidebar.tsx:152]: this hash `#crashreport-sidequest` is how we tell we're coming from the issue details tab

File: static/app/views/userFeedback/userFeedbackEmpty.tsx [hunk 2]
```diff
@@ -84,14 +87,25 @@ export function UserFeedbackEmpty({projectIds}: Props) {
...
       <ButtonList gap={1}>
-        <Button
-          external
-          priority="primary"
-          onClick={() => trackAnalyticsInternal('user_feedback.docs_clicked')}
-          href="https://docs.sentry.io/product/user-feedback/"
-        >
-          {t('Read the docs')}
-        </Button>
+        {hasNewOnboarding ? (
```
Comment (michellewzhang) [inline static/app/views/userFeedback/userFeedbackEmpty.tsx:90]: adds a new button in issue details UF empty state

File: static/app/components/feedback/useFeedbackOnboarding.tsx [hunk 3]
```diff
@@ -24,8 +24,12 @@ export function useFeedbackOnboardingSidebarPanel() {
   const organization = useOrganization();
 
   useEffect(() => {
-    if (location.hash === '#feedback-sidequest') {
+    if (
+      location.hash === '#feedback-sidequest' ||
+      location.hash === '#crashreport-sidequest'
+    ) {
       SidebarPanelStore.activatePanel(SidebarPanelKey.FEEDBACK_ONBOARDING);
+      // this tracks clicks from both feedback index and issue details feedback tab
```
Comment (bruno-garcia) [inline static/app/components/feedback/useFeedbackOnboarding.tsx:32]: do we need to split it up in two? cc @jas-kas

File: static/app/components/feedback/feedbackOnboarding/sidebar.tsx (hunk 1 above)
Comment (bruno-garcia) [inline static/app/components/feedback/feedbackOnboarding/sidebar.tsx:152]: how about `#crashreport-sidequest` also goes into a `const` so we can refer to it here and on a couple of spots on `static/app/components/feedback/useFeedbackOnboarding.tsx` ?

File: static/app/components/onboarding/gettingStartedDoc/utils/feedbackOnboarding.tsx [hunk 4]
```diff
@@ -82,3 +82,44 @@ export const getFeedbackConfigOptions = ({
...
 };
+
+export const getCrashReportModalIntroduction = () =>
+  t(
+    'Collect feedback on your errors by installing our crash-report modal. This allows users to submit feedback after they experience an error via an automatic modal that pops up after an error occurs. The default modal will prompt the user for their name, email address, and description of what occurred.'
+  );
+
+export const getCrashReportModalInstallDescriptionJavaScript = () =>
+  tct(
+    'You can collect feedback right before the event is sent, using [code:beforeSend].',
```
Comment (bruno-garcia) [inline static/app/components/onboarding/gettingStartedDoc/utils/feedbackOnboarding.tsx:93]: technically the event is already sent by the time the modal shows up. Should reword slightly?

```suggestion
    'You can collect feedback at the time the event is sent, using [code:beforeSend].',
```

**Review by bruno-garcia — COMMENTED** [review]
Do we need some tests to verify the right onboarding bit shows up at the right time? Or other relevant parts?

File: static/app/components/feedback/useFeedbackOnboarding.tsx (hunk 3 above)
Comment (michellewzhang) [reply static/app/components/feedback/useFeedbackOnboarding.tsx:32]: i added analytics for the separate buttons as well in this PR; this one just tracks both, so we have all 3 available to us

**Comment by michellewzhang** [comment]
> Do we need some tests to verify the right onboarding bit shows up at the right time? Or other relevant parts?

@bruno-garcia i think we could add a test, but it's not totally necessary imo because the window hash resets between page loads, so there won't be a point where the wrong hash shows up on an unintended page. we're also only setting the hash to be `#crashreport-sidequest` in one spot -- on issue details, and similarly only setting the hash to be `#feedback-sidequest` in one spot -- the feedback index

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**